internal/rbd/rbd_util.go

```diff
@@ -1227,7 +1227,7 @@ func GenVolFromVolID(
 	if mapping != nil {
 		rbdVol, vErr := generateVolumeFromMapping(ctx, mapping, volumeID, vi, cr, secrets)
 		if !errors.Is(vErr, util.ErrKeyNotFound) && !errors.Is(vErr, util.ErrPoolNotFound) &&
-			!errors.Is(vErr, ErrImageNotFound) {
+			!errors.Is(vErr, ErrImageNotFound) && !errors.Is(vErr, rados.ErrPermissionDenied) {
```

Comment:
Repeating the same error check a couple of times... Please put it in a small helper function with a good description of when to use it.

Reply:
Yes i refactored a code a bit with helper function and it should be good now.